cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/BalancingAction.java

```diff
@@ -127,41 +129,29 @@ private BalancingAction(TopicPartition sourceTp,
   private void validate() {
     switch (_actionType) {
       case REPLICA_ADDITION:
-        if (_destinationBrokerId == null) {
-          throw new IllegalArgumentException("The destination broker cannot be null for balancing action " + this);
-        } else if (_sourceBrokerId != null) {
+        validateNotNull(_destinationBrokerId, "The destination broker cannot be null for balancing action " + this);
```

Comment:
I think this is exactly the case where the lazy-evaluation `validateNotNull(T obj, Supplier<String> errorMsgSupplier)` method needs to be used since in the previous version, `"The destination broker cannot be null for balancing action " + this` does not get executed unless `_destinationBrokerId == null` is true. So, the change should be:

validateNotNull(_destinationBrokerId, () -> "The destination broker cannot be null for balancing action " + this);